extension/maxminddb.c

```diff
@@ -389,6 +408,200 @@ static void Reader_dealloc(PyObject *self) {
     PyObject_Del(self);
 }
 
+static PyObject *Reader_iter(PyObject *reader) {
+    ReaderIter_obj *ri = PyObject_New(ReaderIter_obj, &ReaderIter_Type);
+    if (ri == NULL) {
+        return NULL;
+    }
+
+    ri->reader = (Reader_obj *)reader;
+    if (ri->reader->closed == Py_True) {
+        PyErr_SetString(PyExc_ValueError,
+                        "Attempt to iterate over a closed MaxMind DB.");
+        return NULL;
```

Comment:
Should we be freeing any memory on `ri` in this case? And below when we return null I suppose. Not sure.

Reply:
Yeah, good catch. We need to decrement the reference counter.